### PR TITLE
[GCP] support multiple regions when gathering metrics

### PR DESCRIPTION
**Pull request by endorama**
## What does this PR do?

Add `regions` parameter to data streams.

It also updates Kibana min version to 8.5.0, as Metricbeat 8.5.0 is required for the underlying feature: https://github.com/elastic/beats/pull/32964

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

`regions` setting should be added to these data streams:
- [x] compute
- [x] dataproc
- [x] firestore
- [x] gke
- [x] loadbalancing_metrics
- [x] redis
- [x] storage

## Related issues

-

### CODE REVIEW

**Comment by endorama** [comment]
I would merge this after the feature freeze for 8.5.0, so we don't block other possible changes compatible with 8.4.0.

File: packages/gcp/manifest.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ categories:
   - google_cloud
   - cloud
 conditions:
-  kibana.version: ^7.17.6 || ^8.3.0
+  kibana.version: ^7.17.6 || ^8.5.0
```
Comment (kaiyan-sheng) [inline packages/gcp/manifest.yml:18]: Since this PR depends on 8.5.0 release of Metricbeat, should we wait till 8.5.0 release first before merging this one? Also did the Metricbeat PR backported into 7.17.x?

Comment (endorama) [reply packages/gcp/manifest.yml:18]: > Since this PR depends on 8.5.0 release of Metricbeat, should we wait till 8.5.0 release first before merging this one? 

Yes!

> Also did the Metricbeat PR backported into 7.17.x?

It was not, and the policy does not require us to. I'm going to remove Kibana 1.17.x support.

**Comment by tommyers-elastic** [comment]
hey @endorama - one comment from me - i think it's kinda confusing seeing `region` _and_ `regions` in the config. shouldn't we simply replace `region` with `regions`?

Screenshot 2022-09-20 at 13 37 40

**Comment by endorama** [comment]
@tommyers-elastic I agree. The reason behind this is "removing `region` is a breaking change", but I did not consider that we can remove it from the UI without removing it from the `gcp.metrics` module. Less confusing UI, same features. What do you think?

**Comment by endorama** [comment]
I went on and removed `region` setting from data stream with `regions`

**Comment by endorama** [comment]
I tested the upgrade path from a previous version to this one.  
I added `region` to the configuration and upgraded and this is the result:
- `region` is not an hidden field in a policy
- data collection still works, but the policy set `region` cannot longer be changed
- `regions` can be set as expected

Due to this, users that are using `region` today would not be able to edit policies any more after upgrade. Data collection works and the workaround could be to create a new policy and unassign-reassign the agents to the new policy, but I'm not sure if this is ok from the user experience point of view.

**Comment by endorama** [comment]
TL;DR: I'm going to reintroduce `region` with a message about the deprecation.

We had an internal discussion about this and we don't yet have a way to deprecate fields. Due to the upgrade issues and this missing functionality we all agreed that is best not to remove `region` to prevent what could be considered a breaking change.

**Review by aspacca — APPROVED** [review]
LGTM

File: packages/gcp/data_stream/compute/manifest.yml [hunk 2]
```diff
@@ -14,9 +14,17 @@ streams:
       - name: region
         type: text
         title: GCP Region
+        description: deprecated, use regions instead
         multi: false
         required: false
         show_user: true
```
Comment (kaiyan-sheng) [inline packages/gcp/data_stream/compute/manifest.yml:20]: Should we still make the old `region` config parameter `show_user: true`?

Comment (endorama) [reply packages/gcp/data_stream/compute/manifest.yml:20]: I think yes to help with the migration. We should probably hide it in the future, but if we do it with the first release where we introduce `regions` the user will not see the deprecation message in any way.

Comment (kaiyan-sheng) [reply packages/gcp/data_stream/compute/manifest.yml:20]: Sounds good to me. Thanks!

**Review by kaiyan-sheng — APPROVED** [review]
